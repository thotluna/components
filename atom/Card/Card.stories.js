import React from 'react'
import Card from './Card'

export default {
  title: 'atomic/Card',
  component: Card,
}

const Template = (args) => <Card {...args}>Hello Component</Card>

export const Primary = Template.bind({})
Primary.args = {
  primary: true,
  label: 'Component',
}
