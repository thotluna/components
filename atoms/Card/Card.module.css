.card {
  display: flex;
  width: 100%;
  align-items: center;
  padding: 12px 10px;
  border: var(--border-width-thin) solid var(--color-base-transparent);
  border-radius: var(--card-border-radius);
  box-shadow: var(--box-shadow-sm);
  color: var(--color-font-base);
  font-size: var(--font-size-md);
  transition: box-shadow 0.2s ease-in;
}

.size-sm {
  min-height: 45px;
}

.size-md {
  min-height: 80px;
}

.size-lg {
  min-height: 115px;
}

.color-primary {
  background: var(--color-primary);
  color: var(--color-font-inverted);
}

.color-secondary {
  background: var(--color-secondary);
  color: var(--color-font-inverted);
}

.color-base {
  border: var(--border-width-thin) solid var(--color-gray-300);
  background: var(--color-background);
  color: var(--color-font-base);
}

.is-clickable {
  cursor: pointer;
}

.is-draggable {
  cursor: grab;
  user-select: none;
}

.card.is-clickable:active,
.card.is-draggable:active {
  box-shadow: var(--box-shadow-xs);
}

.card:focus {
  box-shadow: var(--box-shadow-focus);
}
