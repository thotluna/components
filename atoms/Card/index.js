export { default } from './Card'
export { options } from './constants'
export { default as styles } from './Card.module.css'
