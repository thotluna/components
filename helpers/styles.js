import classNames from 'classnames'

export const getDynamicClasses = (cssModule = {}, props = {}, classes) => {
  return classes.reduce((classesObject, classKey) => {
    const propValue = props[classKey]
    const className = cssModule[`${classKey}-${propValue}`]

    return className && propValue
      ? { ...classesObject, [className]: propValue }
      : classesObject
  }, {})
}

export const getModuleClasses = (cssModule, classKey) => {
  return (cssModule && cssModule[classKey]) || classKey
}

export const getObjectClasses = (cssModule, object) => {
  return Object.keys(object).reduce((classes, classKey) => {
    const className = cssModule[classKey]
    return className ? { ...classes, [className]: object[classKey] } : classes
  }, {})
}

export const getClasses = (cssModule = {}) => (props) => (...args) => {
  const allClass = args.map((arg) => {
    if (Array.isArray(arg)) {
      const dinamicClasses = getDynamicClasses(cssModule, props, arg)
      // console.log('>>>dinamic', dinamicClasses)
      return dinamicClasses
    } else if (typeof arg === 'string') {
      const moduleClass = getModuleClasses(cssModule, arg)
      // console.log('>>>module', moduleClass)
      return moduleClass
    } else if (typeof arg === 'object') {
      const objectClass = getObjectClasses(cssModule, arg)
      // console.log('>>>object', objectClass)
      return objectClass
    }
  })
  // console.log(allClass)

  const result = classNames(allClass)

  // console.log(result)
  return result
}
