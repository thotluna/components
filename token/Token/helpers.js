import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
