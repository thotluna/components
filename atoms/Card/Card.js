import React from 'react'
import PropTypes from 'prop-types'

import styles from './Card.module.css'
import { options } from './constants'

import { getClasses } from '../../helpers/styles'

export const Card = ({
  color,
  size,
  onClick,
  isClickable,
  isDraggable,
  children,
}) => {
  const getStyles = getClasses(styles)({ color, size })
  return (
    <div
      onClick={onClick}
      className={getStyles('card', ['color', 'size'], {
        'is-clickable': isClickable,
        'is-draggable': isDraggable,
      })}
      tabIndex={isClickable ? 0 : -1}
    >
      {children}
    </div>
  )
}

Card.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  color: PropTypes.oneOf(options.colors),
  size: PropTypes.oneOf(options.sizes),
  isClickable: PropTypes.bool,
  isDraggable: PropTypes.bool,
}

Card.defaultProps = {
  color: 'base',
  size: 'sm',
}

export default Card
