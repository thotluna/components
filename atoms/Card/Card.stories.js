import React from 'react'
import Card, { options } from '.'
import { getVariusTemplate, getOptionsArgTypes } from '../../helpers/storybook'

const Template = (args) => <Card {...args} />
const ListTemplate = getVariusTemplate(Card)

export default {
  title: 'Atoms/Card',
  component: Card,
  args: {
    children:
      'Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!',
  },
  argTypes: {
    color: getOptionsArgTypes(options.colors),
    size: getOptionsArgTypes(options.sizes),
    children: { control: 'text' },
    onClick: { table: { disable: true } },
  },
}

export const Default = Template.bind({})

export const Clickable = Template.bind({})
Clickable.args = {
  isClickable: true,
}

export const Dragabble = Template.bind({})
Dragabble.args = {
  isDraggable: true,
}

export const Colors = ListTemplate.bind({})
Colors.args = { items: options.colors.map((color) => ({ color })) }

export const Sizes = ListTemplate.bind({})
Sizes.args = { items: options.sizes.map((size) => ({ size })) }
