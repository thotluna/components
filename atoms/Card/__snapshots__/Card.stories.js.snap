// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Card Clickable 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="card color-base size-sm is-clickable"
    tabIndex={0}
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
</div>
`;

exports[`Storyshots Atoms/Card Colors 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="card color-base size-sm"
    tabIndex={-1}
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
  <div
    className="card color-primary size-sm"
    tabIndex={-1}
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
  <div
    className="card color-secondary size-sm"
    tabIndex={-1}
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
</div>
`;

exports[`Storyshots Atoms/Card Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="card color-base size-sm"
    tabIndex={-1}
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
</div>
`;

exports[`Storyshots Atoms/Card Dragabble 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="card color-base size-sm is-draggable"
    tabIndex={-1}
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
</div>
`;

exports[`Storyshots Atoms/Card Sizes 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="card color-base size-sm"
    tabIndex={-1}
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
  <div
    className="card color-base size-md"
    tabIndex={-1}
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
  <div
    className="card color-base size-lg"
    tabIndex={-1}
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
</div>
`;
